powersimdata/network/model.py

```diff
@@ -34,75 +35,57 @@ def __init__(self, model, interconnect=None):
 
     def area_to_loadzone(self, *args, **kwargs):
         """Map the query area to a list of loadzones, using the known grid model."""
-        return area_to_loadzone(
-            self.model, *args, mappings=self.zones["mappings"], **kwargs
-        )
+        return area_to_loadzone(self.model, *args, **kwargs)
 
 
-def area_to_loadzone(grid_model, area, area_type=None, mappings=None):
+def area_to_loadzone(model, area, area_type=None):
     """Map the query area to a list of loadzones.
 
-    :param str grid_model: the grid model to use to look up constants for mapping.
-    :param str area: one of: *loadzone*, *state*, *state abbreviation*,
+    :param str model: grid model to use to look up constants for mapping.
+    :param str area: one of *loadzone*, *state*, *state abbreviation*,
         *interconnect*, *'all'*.
-    :param str area_type: one of: *'loadzone'*, *'state'*, *'state_abbr'*,
-        *'interconnect'*.
-    :param iterable mappings: a set of strings, representing area types to use to map.
-        If None, all mappings are tried.
-    :return: (*set*) -- set of loadzone names associated with the query area.
-    :raises TypeError: if area is not None or str.
-    :raises ValueError: if area is invalid or the combination of area and area_type is
-        invalid.
-    :raises KeyError: if a mapping is provided which isn't present for a grid_model.
-
-    .. note:: if area_type is not specified, the function will check the area in the
-        order of 'state', 'loadzone', 'state abbreviation', 'interconnect' and 'all'.
+    :param str area_type: one of *'loadzone'*, *'state'*/*'country'*,
+        *'state_abbr'*/'*country_abbr*', *'interconnect'*. If None, ``area`` will be
+        searched successively into *'state'*/*'country'*, *'loadzone'*,
+        *'state abbreviation'*/*'country abbreviation'*, *'interconnect'* and *'all'*.
+    :return: (*set*) -- set of loadzone names located in the query area.
+    :raises TypeError:
+        if ``area`` is not a str.
+        if ``area_type`` is not None or str.
+    :raises ValueError:
+        if ``area`` is invalid
+        if combination of ``area`` and ``area_type`` is invalid.
     """
+    zones = ModelImmutables(model).zones
+    mappings = zones["mappings"]
 
-    def raise_invalid_area(area_type):
-        raise ValueError("Invalid area for area_type=%s" % area_type)
-
-    zones = ModelImmutables(grid_model).zones
-    mappings = {"loadzone", "state", "state_abbr", "interconnect"}
+    if not isinstance(area, str):
+        raise TypeError("area must be a str")
 
     if area_type is not None and not isinstance(area_type, str):
-        raise TypeError("'area_type' should be either None or str.")
+        raise TypeError("area_type must be either None or str")
+
+    division = [a for a in mappings if "abbr" in a][0].split("_")[0]
+    area2loadzone = {
+        f"{division}": lambda x: zones[f"{division}2loadzone"][x],
```

Comment:
Is `lambda x: zones[f"{division}2loadzone"][x]` equivalent to `zones[f"{division}2loadzone"].get`? If so, I think we could also simplify the values for `f"{division}_abbr"` and `"interconnect"`.

Reply:
Done